src/ucp/core/ucp_mm.c

```diff
@@ -322,9 +322,9 @@ static ucs_status_t ucp_mem_map_common(ucp_context_h context, void *address,
     } else {
         ucs_debug("registering user memory at %p length %zu", address, length);
         memh->alloc_method = UCT_ALLOC_METHOD_LAST;
-        memh->mem_type     = UCS_MEMORY_TYPE_HOST;
         memh->alloc_md     = NULL;
         memh->md_map       = 0;
+        ucp_memory_type_detect_mds(context, address, length, &memh->mem_type);
```

Comment:
minor: maybe do this before line 323, and add memtype print to the ucs_debug message above?